react/Menu/index.jsx

```diff
@@ -132,16 +162,91 @@ class Menu extends Component {
   }
 }
 
+class IternalMenuItem extends Component {
+  constructor() {
+    super()
+    this.element = ''
+    this.state = {
+      style: { display: 'block', visibility: 'hidden', position: 'fixed' }
+    }
+  }
+  //When the component is ready, than let's do some Maths
+  componentDidMount() {
+    this.handleDisplay()
+  }
+
+  handleDisplay() {
+    const { coordinates, itemsStyle, scrollTop } = this.props
+    //Find me & give me my coordinates
+    // eslint-disable-next-line
+    const domItem = ReactDOM.findDOMNode(this.element)
+    const domItemCoordinates = domItem.getBoundingClientRect()
+    //If we have don't have enough space => maths
+    const hasEnoughNotSpace =
+      domItemCoordinates.bottom - scrollTop > window.innerHeight
+    this.setState(
+      {
+        style: {
+          ...itemsStyle,
+          top: hasEnoughNotSpace
+            ? coordinates.top - domItemCoordinates.height
+            : coordinates.top + coordinates.height,
+          left: coordinates.left + coordinates.width - domItemCoordinates.width,
+          visibility: 'hidden',
+          position: 'fixed',
+          display: 'block'
+        }
+      },
+      () => {
+        /* We need callback here, because sometimes, getBoundingClient() 
+            doesn't return the right style... 
+          */
+        const updatedItem = domItem.getBoundingClientRect()
+
+        this.setState({
+          style: {
+            ...itemsStyle,
+            ...this.state.style,
+            top: hasEnoughNotSpace
+              ? coordinates.top - updatedItem.height
+              : coordinates.top + coordinates.height,
+            left: coordinates.left + coordinates.width - updatedItem.width,
+            visibility: 'visible'
+          }
+        })
+      }
+    )
+  }
+
+  render() {
+    return (
+      <div
+        className={cx(styles['c-menu__inner'], styles['c-menu__inner--opened'])}
+        ref={element => {
+          if (element !== null && element !== this.element) {
+            this.element = element
+          }
+        }}
+        style={({ ...this.props.itemsStyle }, { ...this.state.style })}
+      >
+        {this.props.renderItems()}
+      </div>
+    )
+  }
+}
 Menu.defaultProps = {
   position: 'left',
   disabled: false,
   component: null,
   onSelect: null,
-  onSelectDisabled: null
+  onSelectDisabled: null,
+  itemsStyle: {}
 }
 
 Menu.propTypes = {
-  /** Use position='right' to display the menu to the right */
+  /** Use position='right' to display the menu to the right. Use left to display it to the left.
+   * Use position='fixed' to let it choose if it has enough space to be displayed at bottom.
```

Comment:
I think `position` should not be used here. The thing we are trying to achieve here is an "adjustment" of the position based on viewport estate.

I think this adjustment should be done automatically without interaction from the user, no additional property is needed IMHO.

The position attribute is ortogonal to this since you first choose how you would like the popover to appear, maybe with a preference on right or left, given that it has all the space available, and then you adjust.

Reply:
The menu is used every where in our code base. I don't want to break anything, specially I don't deal with scrolling / resizing issue. But you're right, I shouldn't use `position`.

I'll add a new property : `popover` {true | false}, and let `position`. I can also add a `console.warn` if popover is set to default / false ?